pages/price-feeds/use-real-time-data/near.mdx

```diff
@@ -9,83 +9,315 @@ price feeds, handling the retrieval and updating of price data.
-   deposit will be refunded, but you can calculate an esimtate by calling
-   the `get_update_fee_estimate` method against the Pyth contract.
-
-   Fetching a price feed is similar:
-
-   ```js
-   async function fetchPriceFeed() {
-     const near = await connect(config);
-     const account = await near.account();
-     const contractId = "pyth-oracle.testnet";
-     const identifier = "PriceIdentifier";
-
-     const priceFeed = await account.viewFunction(
-       contractId,
-       "get_price",
-       args: { identifier }
-     );
-
-     console.log("Price Feed Data: ", priceFeed);
-   }
-   ```
+price information.
+
+For a full overview of methods provided by the NEAR
+contract, see [the interface](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/near/receiver/src/ext.rs)] exposed by the receiver contract.
+
+## Getting Started
+
+To get started with Pyth oracle you will need to gather the following information which differ between networks:
+
+- Price ID(s)
+- HermesAPI Endpoint
+- Smart contract address
+
+| Network   | Price Feed IDs                                                                                   | Hermes API Address         | Contract Address                                                                 |
+| --------- | ------------------------------------------------------------------------------------------------ | -------------------------- | -------------------------------------------------------------------------------- |
+| `testnet` | [NEAR `testnet` Price Feed IDs](https://www.pyth.network/developers/price-feed-ids#near-testnet) | `hermes-beta.pyth.network` | [pyth-oracle.testnet](https://testnet.nearblocks.io/address/pyth-oracle.testnet) |
+| `mainnet` | [NEAR `mainnet` Price Feed IDs](https://www.pyth.network/developers/price-feed-ids#near-mainnet) | `hermes.pyth.network`      | [pyth-oracle.near](https://nearblocks.io/address/pyth-oracle.near)               |
+
+Note: When using Price Feed IDs, you will need to remove the `0x` prefix.
+
+---
+
+### `update_price_feeds`
+
+> Updates the Pyth Oracle contract data with the price feed you provide.
+
+- args: `data` _(off-chain hex-encoded price feed)_
+- type: `object`
+- example: `{ "data": "504e41...' }`
+
+Update the Pyth Oracle contract with new price feed data in two main steps:
+
+1. [Fetch off-chain price feed](#1-fetch-off-chain-price-feed)
+2. [Update Pyth Oracle contract with off-chain price feed](#2-update-pyth-oracle-contract-price-feed)
+
+#### 1) Fetch off-chain price feed
+
+You can obtain an off-chain price feed using Pyth's [Hermes API](https://hermes-beta.pyth.network/docs/).
+
+To use these endpoints, you will need to provide a Price Feed ID and ensure you are targeting the correct network. See [Getting Started](#getting-started) for more information.
+
+Here is a node.js example of fetching the latest price feed using `/v2/updates/price/latest` endpoint:
+
+`Example:`
+
+```js
+const axios = require("axios");
```

Comment:
we can use the Hermes SDK here.

Reply:
I just tried [this](https://github.com/pyth-network/pyth-crosschain/blob/main/apps/hermes/client/js/src/examples/HermesClient.ts) example with it and it worked. We will try it separately later and fix any issues that arise.